controller/src/main/java/org/jboss/as/controller/ProxyStepHandler.java

```diff
@@ -267,6 +276,20 @@ public void handleResult(OperationContext.ResultAction resultAction, OperationCo
         }
     }
 
+    private ModelNode processResponseHeaders(ModelNode responseHeaders) {
+        if (!responseHeaders.hasDefined(ACCESS_CONTROL) || !forServer) {
+            return responseHeaders;
+        } else {
+            ModelNode result = responseHeaders.clone();
+            for (ModelNode accItem : result.get(ACCESS_CONTROL).asList()) {
+                ModelNode itemAddrNode = accItem.get("absolute-address");
```

Comment:
I actually have that on my TODO list, but I wanted to get this done and also figured this patch was big enough! I'll do it at some point pretty soon though.